cores/esp32/esp32-hal-timer.c

```diff
@@ -25,6 +25,8 @@
 #define HWTIMER_LOCK()      portENTER_CRITICAL(timer->lock)
 #define HWTIMER_UNLOCK()    portEXIT_CRITICAL(timer->lock)
 
+portMUX_TYPE arduino_mux = portMUX_INITIALIZER_UNLOCKED;
+
```

Comment:
first mayb e move those to esp32-hal.h and esp32-hal-misc.c ? should not be in the timers ;)

Reply:
Perhaps also rename arduino_mux to interrupt_lock or interrupt_mux ?